src/hotspot/share/classfile/classFileError.cpp

```diff
@@ -92,6 +92,12 @@ void ClassFileParser::classfile_icce_error(const char* msg,
                      msg, _class_name->as_klass_external_name(), k->external_name());
 }
 
+void ClassFileParser::classfile_icce_error(const char* msg,
+                                           TRAPS) const {
+  ResourceMark rm(THREAD);
```

Comment:
I don't know what this ResourceMark does since you allocate resources in the caller, not here.

Reply:
It follows the same pattern as all the other throwing methods - the RM is for the `_class_name->as_C_string()`. I think you may be right that it is the callers that should have the RM. Can I file a follow up RFE to look at that?

Reply:
Now I see what you mean. Fixed.